### ISSUE TITLE
Don't rely on file creation timestamps

### ISSUE THREAD

**Issue by jamilbk** [tokio-rs/tracing#2918]
### Crates

tracing_appender

### Motivation

Apple's new Privacy regulations force developers to now declare a "Privacy Manifest" file
with details on why the API in question is needed.

This creates a hurdle for building applications for Apple platforms that use tracing_appender as it reads file timestamps by default.

[See the error message we received from the App Review team](https://github.com/firezone/firezone/issues/4377#issuecomment-2026189479).

### Proposal

I suppose there's not a great solution for this. In our case, the filenames themselves have the creation timestamp in them, so tracing_appender may be able to use that.

Otherwise, perhaps exposing an option to prevent hitting this codepath would be sufficient.

### Alternatives

We're considering just ripping out tracing_appender altogether and just live with one big logfile.

**Comment by davidbarsky** [comment]
`.metadata().ok()` is an API coming from `std`; which makes things... complicated. I'd recommend making your own appender that is compliant with Apple's privacy manifests.

**Comment by jamilbk** [comment]
Yeah, we ended up just submitting a PrivacyManifest because even `std` pulls in "offending" API calls.
